src/hashcat.c

```diff
@@ -1194,6 +1201,50 @@ int hashcat_session_execute (hashcat_ctx_t *hashcat_ctx)
 
     user_options->quiet = false;
   }
+  else if (user_options->guess_hash_mode == true)
+  {
+    u64 rc = 0;
+
+    rc_final = 1;
+
+    event_log_info (hashcat_ctx, "Starting guessing mode...\n");
+
+    user_options->quiet = true;
+
+    if (user_options->hash_mode_chgd == true)
+    {
+      if (outer_loop (hashcat_ctx) != -1) rc++;
+    }
+    else
+    {
+      char *modulefile = (char *) hcmalloc (HCBUFSIZ_TINY);
+
+      for (int i = 0; i < MODULE_HASH_MODES_MAXIMUM; i++)
+      {
+        // skip STDOUT, Plaintext and Password Safe v2 modes
+        if (i == 2000 || i == 99999 || i == 9000) continue;
```

Comment:
There is no better way to do this? I think this is not maintainable in long term.

Reply:
There are a thousand ways to make everything better.
First I always make a prototype, then I improve it ;)

Reply:
Okay, I thought that it's a final version ;)

Anyway, I am still learning Hashcat source code but maybe a new field `module_flags` in `module_ctx_t` struct would be a nice solution.
If flag `MODULE_GUESS_MODE_SKIP` (or something similar) is set then skip.

Anyway, nice feature!